bats_test/writing_tests.md

```diff
@@ -0,0 +1,64 @@
+# Writing BATS tests
+
+Some guidelines on writing your own BATS tests
+## Globals
+
+There are several global variables you can use to introspect on Bats tests:
+
+- $BATS_TEST_FILENAME is the fully expanded path to the Bats test file.
+
+- $BATS_TEST_DIRNAME is the directory in which the Bats test file is located.
+
+- $BATS_TEST_NAMES is an array of function names for each test case.
+
+- $BATS_TEST_NAME is the name of the function containing the current test case.
+
+- $BATS_TEST_DESCRIPTION is the description of the current test case.
+
+- $BATS_TEST_NUMBER is the (1-based) index of the current test case in the test file.
+
+- $BATS_TMPDIR is the location to a directory that may be used to store temporary files.
+
+- $BATS_RUN_COMMAND string contains the command and command arguments passed to *run* 
+  
+Check every variable [here](https://bats-core.readthedocs.io/en/stable/writing-tests.html#special-variables)
+## Support libs
+
+[Support](https://github.com/bats-core/bats-support#bats-support)
+
+[Files](https://github.com/bats-core/bats-file#index-of-all-functions)
+
+[Assertions](https://github.com/bats-core/bats-assert#usage)
+
+
+## Basic Commands
+
+### Log to console
+
+Always  ``` echo 'text' >&3 ```
```

Comment:
Would you mind specifying which STD stream does `>&3` stand for? Is it the error stream?

Reply:
Its on BATS documentation. Its a way to log directly https://bats-core.readthedocs.io/en/stable/writing-tests.html#file-descriptor-3-read-this-if-bats-hangs